docs/shared/mpas.yaml

```diff
@@ -44,11 +44,18 @@ mpas:
     validate: true
   run_dir: /path/to/run/directory
   streams:
-    path: "{{ user.mpas_app }}/MPAS-Model/streams.atmosphere"
-    values:
-      input_filename: conus.init.nc
-      input_interval: "06:00:00"
-      output_interval: "06:00:00"
+    - name: input
+      type: input
+      filename_template: conus.init.nc
+      input_interval: initial_only
+      mutable: false
+    - name: output
+      type: output
+      filename_template: history.$Y-$M-$D_$h.$m.$s.nc
+      output_interval: "6:00:00"
+      mutable: true
+      files:
+        - stream_list.atmosphere.output
```

Comment:
See the update UW YAML docs for this block and/or the new `mpas_streams.jsonschema` for background on this.

Reply:
> losing our ability to update the values contained in the lists with a call to `update_values()`

There is no `update_values` in the current implementation, so there's nothing to lose: Streams are defined in UW YAML (like Rocoto workflows are defined in UW YAML) and serialized as XML, so there's nothing to update.

That said, I see from another comment that you might be asking for the ability to read and modify an existing streams file, which would make the concern pertinent again. I don't think `list` or `dict` matters, because we'd have to use XPath queries (or an equivalent mechanism) to select and update elements matching appropriate criteria; but `dict` might still be more appropriate because, although XML is not a mapping-like format (it's a sequence-like format, e.g. there's no language rule against identical repeated elements like `<food><fruit name="banana"/><fruit name="banana"/></food>` as there's nothing wrong with a `list` containing `["banana", "banana"]`), MPAS does request -- and maybe require, but I haven't checked -- that the `name` attribute on a stream element be unique. And element order is not meaningful in streams files, as far as I know.

I don't see `io_type` in the streams files I got from jet but, as mentioned in another comment, we could add support for any or all of the elements and attributes that MPAS supports.

But we should probably decide first if we want to continue with this kind of update. XML has different characteristics than the other config formats we deal with and could have some unexpected behaviors. Generating it from validated config is straightforward, but updating it programmatically could yield some surprises, given that the use cases are fairly new to us. As of this writing, I think that generating it is the safer and more reliable approach, but I think both ways are possible.

Reply:
Updates have been made so that the `streams` block is now a mapping from strings (stream names) to mappings of key-value config pairs. Support for the full range of MPAS-supported stream configuration options has been added.